cellar_rebalancer/src/cellars/uniswapv3.rs

```diff
@@ -158,11 +166,44 @@ impl<T: 'static + Middleware> UniswapV3CellarState<T> {
     }
 }
 
-// Struct for UniswapV3CellarTickInfo
-#[derive(Clone, Debug)]
-pub struct UniswapV3CellarTickInfo {
-    pub(crate) token_id: U256,
-    pub(crate) tick_upper: i32,
-    pub(crate) tick_lower: i32,
-    pub(crate) weight: u32,
+pub struct UniswapV3CellarAllocator;
+
+#[async_trait]
+impl server::UniswapV3CellarAllocator for UniswapV3CellarAllocator {
+    async fn rebalance(
+        &self,
+        request: tonic::Request<RebalanceRequest>,
+    ) -> Result<tonic::Response<RebalanceResponse>, tonic::Status> {
+        let request = request.get_ref();
+        let tick_ranges: Vec<proto::TickRange> = request
+            .data
+            .clone()
+            .into_iter()
+            .map(|d| proto::TickRange {
+                upper: d.upper_price,
+                lower: d.lower_price,
+                weight: d.weight,
+            })
+            .collect();
+        let pair_id = APP
+            .config()
+            .cellars
+            // sketchy
+            .get(0)
+            .unwrap()
```

Comment:
If there is an error here, we should return one by setting our return `tonic::Status` appropriately.